rad/src/lib.rs

```diff
@@ -242,6 +264,106 @@ async fn rng_response(
     Ok(RadonReport::from_result(Ok(random_bytes), context))
 }
 
+/// Handle HTTP-POST response
+async fn http_post_response(
```

Comment:
also here, could be combined with the http_get?

Reply:
This one has a different line:

    let request = surf::post(&retrieve.url).body(retrieve.body.clone());

I guess we can add a boolean flag...